File: Leetcode/typescript/921-min-add-valid-paren.ts
function minAddToMakeValid(s: string): number {
  let openCount = 0;
  let res = 0;

  for (const c of s) {
    if (c === "(") {
      openCount++;
    } else {
      if (openCount === 0) res++;
      openCount = Math.max(openCount - 1, 0);
    }
  }
  return res + openCount;
}

File: Leetcode/python/921-min-add-valid-paren.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        openCount = 0
        res = 0

        for c in s:
            if c == "(":
                openCount += 1
            else:
                if openCount == 0:
                    res += 1
                openCount = max(openCount - 1, 0)


        return res + openCount
